lighthouse-core/gather/gatherers/full-page-screenshot.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * @license Copyright 2020 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('./gatherer.js');
+
+// JPEG quality setting
+// Exploration and examples of reports using different quality settings: https://docs.google.com/document/d/1ZSffucIca9XDW2eEwfoevrk-OTl7WQFeMf0CgeJAA8M/edit#
+const FULL_PAGE_SCREENSHOT_QUALITY = 30;
+// Maximum screenshot height in Chrome https://bugs.chromium.org/p/chromium/issues/detail?id=770769
+const MAX_SCREENSHOT_HEIGHT = 16384;
+// Maximum data URL size in Chrome https://bugs.chromium.org/p/chromium/issues/detail?id=69227
+const MAX_DATA_URL_SIZE = 2 * 1024 * 1024 - 1;
+
+/**
+ * @param {string} str
+ */
+function snakeCaseToCamelCase(str) {
+  return str.replace(/(-\w)/g, m => m[1].toUpperCase());
+}
+
+class FullPageScreenshot extends Gatherer {
+  /**
+   * @param {LH.Gatherer.PassContext} passContext
+   * @param {number} maxScreenshotHeight
+   * @return {Promise<LH.Artifacts.FullPageScreenshot>}
+   */
+  async _takeScreenshot(passContext, maxScreenshotHeight) {
+    const driver = passContext.driver;
+    const metrics = await driver.sendCommand('Page.getLayoutMetrics');
+    const deviceScaleFactor = await driver.evaluateAsync('window.devicePixelRatio');
+    const width = Math.min(metrics.contentSize.width, maxScreenshotHeight);
+    const height = Math.min(metrics.contentSize.height, maxScreenshotHeight);
+
+    await driver.sendCommand('Emulation.setDeviceMetricsOverride', {
```

Comment:
I compared the typical metrics override params (set in emulation.js) to this one:

![image](https://user-images.githubusercontent.com/39191/87368922-01f49980-c534-11ea-9fd9-f287a50fe533.png)

i dont know if the differences matter, but just pointing out they do exist. feels like we should be consistent.